posthog-core/src/index.ts

```diff
@@ -0,0 +1,556 @@
+    this.flushInterval = options?.flushInterval ?? 10000
+    this.captureMode = options?.captureMode || 'form'
+    this.sendFeatureFlagEvent = options?.sendFeatureFlagEvent ?? true
+    this._decidePollInterval = Math.max(0, options?.decidePollInterval ?? 30000)
+    // If enable is explicitly set to false we override the optout
+    this._optoutOverride = options?.enable === false
+    this._retryOptions = {
+      retryCount: options?.fetchRetryCount ?? 3,
+      retryDelay: options?.fetchRetryDelay ?? 3000,
+    }
+    this._sessionExpirationTimeSeconds = options?.sessionExpirationTimeSeconds ?? 1800 // 30 minutes
+
+    // NOTE: It is important we don't initiate anything in the constructor as some async IO may still be underway on the parent
+    if (options?.preloadFeatureFlags !== false) {
+      safeSetTimeout(() => {
+        void this.reloadFeatureFlagsAsync()
+      }, 1)
+    }
+  }
+
+  protected getCommonEventProperties(): any {
+    const featureFlags = this.getFeatureFlags()
+    return {
+      $lib: this.getLibraryId(),
+      $lib_version: this.getLibraryVersion(),
+      $active_feature_flags: featureFlags ? Object.keys(featureFlags) : undefined,
+      $enabled_feature_flags: featureFlags,
+    }
+  }
+
+  // NOTE: Props are lazy loaded from localstorage hence the complex getter setter logic
+  private get props(): PostHogEventProperties {
+    if (!this._props) {
+      this._props = this.getPersistedProperty<PostHogEventProperties>(PostHogPersistedProperty.Props)
+    }
+    return this._props || {}
+  }
+
+  private set props(val: PostHogEventProperties) {
+    this._props = val
+  }
+
+  private _props: PostHogEventProperties | undefined
+
+  public get optedOut(): boolean {
+    return this.getPersistedProperty(PostHogPersistedProperty.OptedOut) ?? this._optoutOverride ?? false
+  }
+
+  optIn(): void {
+    this.setPersistedProperty(PostHogPersistedProperty.OptedOut, false)
+  }
+
+  optOut(): void {
+    this.setPersistedProperty(PostHogPersistedProperty.OptedOut, true)
+  }
+
+  on(event: string, cb: (e: any) => void): () => void {
+    return this._events.on(event, cb)
+  }
+
+  reset(): void {
+    for (const key in PostHogPersistedProperty) {
+      this.setPersistedProperty((PostHogPersistedProperty as any)[key], null)
+    }
+    this.setPersistedProperty(PostHogPersistedProperty.DistinctId, generateUUID(globalThis))
+  }
+
+  private buildPayload(payload: { event: string; properties?: PostHogEventProperties; distinct_id?: string }): any {
+    return {
+      distinct_id: payload.distinct_id || this.getDistinctId(),
+      event: payload.event,
+      properties: {
+        ...this.props, // Persisted properties first
+        ...(payload.properties || {}), // Followed by user specified properties
+        ...this.getCommonEventProperties(), // Followed by common PH props
+        $session_id: this.getSessionId(),
+      },
+    }
+  }
+
+  getSessionId(): string | undefined {
```

Comment:
I think we're missing a `resetSessionId()` method based on https://github.com/PostHog/posthog/issues/10780

Reply:
Good catch, I'll add it